### 1) Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать (желательно без использования статистических методов наподобие std, var, mean: 
### а) среднее арифметическое
### б) среднее квадратичное отклонение
### в) смещенную и несмещенную оценки дисперсий для данной выборки.

In [1]:
import numpy as np
from math import factorial

In [2]:
arr=np.array([100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150])
arr

array([100,  80,  75,  77,  89,  33,  45,  25,  65,  17,  30,  24,  57,
        55,  70,  75,  65,  84,  90, 150])

a) Расчитаем среднее арифметическое:

In [3]:
def mean_value(array):
    return sum(array)/len(array)

In [4]:
print(f'Среднее арифметическое для данной выборки М(Х) = {mean_value(arr): .2f}')

Среднее арифметическое для данной выборки М(Х) =  65.30


In [5]:
np.mean(arr)

65.3

б) Расчитаем среднее квадратичное отклонение выборки:

In [6]:
def mean_square_deviation(array):
    square_dev=(array-mean_value(array))**2
    return (sum(square_dev)/len(square_dev))**(1/2)

In [7]:
print(f'Среднее квадратичное отклонение для данной выборки SD = {mean_square_deviation(arr): .4f}')

Среднее квадратичное отклонение для данной выборки SD =  30.8239


Проверим с помощью функции std:

In [8]:
np.std(arr)

30.823854398825596

в) Расчитаем смещенную оценку дисперсии:

In [10]:
def dispers(array, no_off=False):
    square_dev=(array-mean_value(array))**2
    return sum(square_dev)/(len(square_dev)-1) if no_off else sum(square_dev)/len(square_dev)

#аргумент no_off отвечает за расчет смещенной, или несмещенной оценки дисперсии. По умолчанио расчитывается смещенная.#

In [11]:
print(f'Смещенная оценка дисперсии для данной выборки D = {dispers(arr): .4f}\n'
      f'Немещенная оценка дисперсии для данной выборки D = {dispers(arr, True): .4f}')

Смещенная оценка дисперсии для данной выборки D =  950.1100
Немещенная оценка дисперсии для данной выборки D =  1000.1158


Проверим с помощью функции var:

In [12]:
np.var(arr)

950.11

In [13]:
np.var(arr, ddof=1)

1000.1157894736842

##### Ответ:
**а) М(Х) = 65.30;**

**б) σ = 30.8239;**

**в) σ_смещ^2 = 950.11, σ_несмещ^2  =1000.1158.**


### 2) В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4. Какова вероятность того, что 3 мяча белые?

Пусть A-искомое событие "из двух корзин вытащили 3 белых" мяча. Тогда это событие может произойти следующими способами:
а) A1 из первой корзины не вытащили ни одного белого мяча, из второй вытащили 3 белых мяча;
б) A2 из первой корзины вытащили 1 белый мяч, из второй вытащили 2 белых мяча;
в) A3 из первой корзины вытащили 2 белых мяча, из второй вытащили 1 белый мяч.
Вероятность события A будет определяться как сумма вероятностей этих событий.

Определим вероятность события A1:
    
P(A) =  (C_3^2)/(C_8^2 )∙(C_5^3∙C_7^1)/(C_12^4)

In [15]:
def combinations(n,k):
    return factorial(n)/(factorial(k)*factorial(n-k))

In [16]:
P1=(combinations(3,2)/combinations(8,2))*(combinations(5,3)*combinations(7,1)/combinations(12,4))
print(f'Вероятность события "из первой корзины не вытащили ни одного белого мяча'
      f', из второй вытащили 3 белых мяча"\nP(A1) = {P1: .4f}')

Вероятность события "из первой корзины не вытащили ни одного белого мяча, из второй вытащили 3 белых мяча"
P(A1) =  0.0152


Определим вероятность события A2:

P(A)=  (C_5^1∙C_3^1)/(C_8^2)∙(C_5^2∙C_7^2)/(C_12^4)

In [21]:
P2=(combinations(5,1)*combinations(3,1)*combinations(5,2)*combinations(7,2))/(combinations(8,2)*combinations(12,4))
print(f'Вероятность события "из первой корзины вытащили 1 белый мяч'
      f', из второй вытащили 2 белых мяча"\nP(A2) = {P2: .4f}')

Вероятность события "из первой корзины вытащили 1 белый мяч, из второй вытащили 2 белых мяча"
P(A2) =  0.2273


Определим вероятность события A3:
    

P(A)=  (C_5^2)/(C_8^2)∙(C_5^1∙C_7^3)/(C_12^4)

In [18]:
P3=(combinations(5,2)*combinations(5,1)*combinations(7,3))/(combinations(8,2)*combinations(12,4))
print(f'Вероятность события "из первой корзины вытащили 2 белых мяча'
      f', из второй вытащили 1 белый мяч"\nP(A2) = {P3: .4f}')

Вероятность события "из первой корзины вытащили 2 белых мяча, из второй вытащили 1 белый мяч"
P(A2) =  0.1263


Вычислим P(A) = P(A1 + A2 + A3) = P(A1) + P(A2) + P(A3):

In [22]:
print(f'Вероятность того, что из вытащенных мячей 3 белые Р(А) = {P1+P2+P3: .4f}')

Вероятность того, что из вытащенных мячей 3 белые Р(А) =  0.3687


##### Ответ:
**P(A) = 0.3687 = 36.87%**

### 3) На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6. 
#### Найти вероятность того, что выстрел произведен: 
#### a) первым спортсменом
#### б) вторым спортсменом
#### в) третьим спортсменом.

Рассмотрим следующие события:

A - в результате выстрела, цель поражена;

B1 выстрел произведен первым спортсменом;

B2 выстрел произведен вторым спортсменом;

B3 выстрел произведен третьим спортсменом.

Вероятности того, что выстрел произведен определенным спортсменом равны, т.к. выстрел 1, и произвести его могут только 3 этих спортсмена, т.е. события B1, B2, B3 составляют полную группу событий.
⇒P(B1)=P(B2)=P(B3)=13,

Вероятность того, что произошло событие A, при условии наступления события Bn даны в условии задачи, т.е.

PB1 (A) = 0.9, PB2 (A) = 0.8, PB3 (A) = 0.6, и, по формуле полной вероятности, вероятность наступления события A  -

P(A)=1/3∙0.9 + 1/3∙0.8 + 1/3∙0.6


In [23]:
PB=1/3
PA=PB*0.9+PB*0.8+PB*0.6
print(f'Полная вероятность наступления события А Р(А) = {PA: .4f}')

Полная вероятность наступления события А Р(А) =  0.7667


Теперь, исходя из того, что событие A уже произошло, с помощью формулы Байеса оценим вероятность того, что имело место одно из событий Bn:

In [24]:
PAB1=PB*0.9/PA
PAB2=PB*0.8/PA
PAB3=PB*0.6/PA
print(f'Вероятность того, что выстрел произвёл первый спортсмен: {PAB1: .4f};\n'
      f'Вероятность того, что выстрел произвёл второй спортсмен: {PAB2: .4f};\n'
      f'Вероятность того, что выстрел произвёл третий спортсмен: {PAB3: .4f}.')

Вероятность того, что выстрел произвёл первый спортсмен:  0.3913;
Вероятность того, что выстрел произвёл второй спортсмен:  0.3478;
Вероятность того, что выстрел произвёл третий спортсмен:  0.2609.


### Ответ:
**а) Вероятность того, что выстрел произвёл первый спортсмен: PA(B1) = 0.3913 = 39.13%;**

**б) Вероятность того, что выстрел произвёл второй спортсмен: PA(B2) = 0.3478 = 34.78%;**

**в) Вероятность того, что выстрел произвёл третий спортсмен:  PA(B3)= 0.2609 = 26.09%.**

### 4) В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе. Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9. Студент сдал первую сессию. 

#### Какова вероятность, что он учится:

#### a) на факультете A 

#### б) на факультете B 

#### в) на факультете C?

Пусть n - количество студентов на факультете A, тогда на факультете B учится n студентов, а на факультете B  2n студентов.
Введем события SA, SB, SC, что сессию сдаёт студент факультетов A, B, и С соответственно, и событие D-сессия сдана.
события SA, SB, SC образуют полную группу событий, ⇒P(SA)+P(SB)+P(SC) = 1

T.e. 1/n+1/n+1/2n= 1n = 2.5⇒ P(SA) = 1/n = 0.25, P(SB) = 1/n = 0.25,  P(SC) = 1/2n = 0.5.

Условные вероятности Наступления события D, при условии что произошло одно из событий Sn даны в условии задачи.
Полная вероятность наступления события D исходя из формулы полной вероятности:
P(D) = 0.25∙0.8+0.25∙0.7+ 0.5∙0.9

In [26]:
PD=0.25*0.8+0.25*0.7+0.5*0.9
print(f'Полная вероятность наступления события D, P(D) = {PD}.')

Полная вероятность наступления события D, P(D) = 0.825.


По формуле Байеса вычислим вероятность того, на каком факультете учится студент, успешно сдавший сессию.

In [28]:
PDSA=0.25*0.8/PD
PDSB=0.25*0.7/PD
PDSC=0.5*0.9/PD
print(f'Вероятность того, что студент учится на факультете А: {PDSA: .4f};\n'
      f'Вероятность того, что студент учится на факультете B: {PDSB: .4f};\n'
      f'Вероятность того, что студент учится на факультете C: {PDSC: .4f}.')

Вероятность того, что студент учится на факультете А:  0.2424;
Вероятность того, что студент учится на факультете B:  0.2121;
Вероятность того, что студент учится на факультете C:  0.5455.


#### Ответ:

**Вероятность того, что студент учится на факультете А: PD(SA)  = 0.2424 = 24.24%;**

**Вероятность того, что студент учится на факультете B: PD(SB)  = 0.2121 = 21.21%;**

**Вероятность того, что студент учится на факультете C: PD(SC)  = 0.5455 = 54.55%.**

### 5) Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25.

#### Какова вероятность того, что в первый месяц выйдут из строя: 

#### а) все детали 

#### б) только две детали

#### в) хотя бы одна деталь 

#### г) от одной до двух деталей?

Введем следующие события:
    
а) Событие A1-вышла из строя первая деталь, по условию задачи-вероятность этого события: P(A1)=0.1;
    
б) Событие Ā1-противоположное событию A1 - первая деталь не вышла из строя, вероятность этого события: P(Ā1)=1−P(A1)=1−0.1=0,9;
    
в) Событие A2-вышла из строя вторая деталь, по условию задачи-вероятность этого события: P(A2)=0.2;
    
г) Событие Ā2-противоположное событию A2 - вторая деталь не вышла из строя, вероятность этого события: P(Ā2)=1−P(A2)=1−0.2=0,8;
    
д) Событие A3-вышла из строя первая деталь, по условию задачи-вероятность этого события: P(A3) =0.25;
    
е) Событие Ā3-противоположное событию A3 - третья деталь не вышла из строя, вероятность этого события: P(Ā3)=1−P(A3)=1−0.25=0,9.


_а) Для того, чтобы произошло событие "вышли из строя все детали", необходимо, чтобы одновременно произошли события A1, A2, и A3. А вероятностью этого события P(3) будет произведением вероятностей этих событий._

P(A)=P(A1)∙P(A2)∙P(A3)=0.1∙0.2∙0.25


In [29]:
P3=0.1*0.2*0.25
print(f'Вероятность того, что из строя выйдут все детали Р(3) = {P3: .4f}')

Вероятность того, что из строя выйдут все детали Р(3) =  0.0050


_б) Для того, чтобы произошло событие "вышли из строя 2 детали", необходимо что бы случилось одна из следующих комбинаций событий:_

_1. Произошли события A1 и A2 и Ā3;_
    
_2. Произошли события A1 и A3 и Ā2;_
    
_3. Произошли события A2 и A3 и Ā1._
    
_А вероятностью этого события P(2) будет суммой вероятностей этих комбинаций событий._

P(2)=P(A1)∙P(A2)∙P(Ā3)+P(A1)∙P(A3)∙P(Ā2)+P(A2)∙P(A3)∙P(Ā1)=0.1∙0.2∙0.75+0.1∙0.25∙0.8+0.2∙0.25∙0.9


In [30]:
P2=0.1*0.2*0.75+0.1*0.25*0.8+0.2*0.25*0.9
print(f'Вероятность того, что из строя выйдут только 2 детали Р(2) = {P2: .4f}')

Вероятность того, что из строя выйдут только 2 детали Р(2) =  0.0800


_в) Событие "выйдет из строя хотя бы одна деталь" можно рассмотреть как противоположное событию "не вышло из строя ни одной детали"._

_Пусть P(0) вероятность события "не вышло из строя ни одной детали". Данное событие может произойти при одновременном наступлении событий Ā1, Ā2, и Ā3._

P(0)=P(Ā1)∙P(Ā2)∙P(Ā3)=0.9∙0.8∙0.75


In [31]:
P0=0.9*0.8*0.75
print(f'Вероятность того, что из строя не выйдет ни одной детали Р(0) = {P0: .4f}')

Вероятность того, что из строя не выйдет ни одной детали Р(0) =  0.5400


Тогда вероятность события "выйдет из строя хотя бы одна деталь"  P(≥1)=1-P(0)

In [33]:
P_0=1-P0
print(f'Вероятность того, что выйдет из строя хотя бы одна деталь Р(>=1) = {P_0: .4f}')

Вероятность того, что выйдет из строя хотя бы одна деталь Р(>=1) =  0.4600


_г) Событие "из строя выйдут от одной до двух деталей" произойдет при наступлении события "из строя выйдет одна деталь" или из строя выйдут 2 детали, следовательно его вероятность будет равна сумме вероятностей этих событий: P(1,2)=P(1)+P(2)
P(2)=0.08, найдем P(1):_

_Для того, чтобы произошло событие "из строя выйдет одна деталь", необходимо что бы случилось одна из следующих комбинаций событий:_

_1. Произошли события A1 и Ā2 и Ā3;_

_2. Произошли события A2 и Ā1 и Ā3;_

_3. Произошли события A3 и Ā1 и Ā2._

_А вероятностью этого события P(1) будет суммой вероятностей этих комбинаций событий._

P(1)=P(A1)∙P(Ā2)∙P(Ā3)+P(A2)∙P(Ā1)∙P(Ā3)+P(A3)∙P(Ā1)∙P(Ā2)=0.1∙0.8∙0.75+0.2∙0.9∙0.75+0.25∙0.9∙0.8


In [34]:
P1=0.1*0.8*0.75+0.2*0.9*0.75+0.25*0.9*0.8
print(f'Вероятность того, что выйдет из строя одна деталь Р(1) = {P1: .4f}')

Вероятность того, что выйдет из строя одна деталь Р(1) =  0.3750


**P(1,2)=P(1)+P(2)=0.375+0.08=0.455**

#### Ответ:

**а) Вероятность того что в первый месяц выйдут из строя все детали P(3) = 0.005 = 0.50%**

**б) Вероятность того что в первый месяц выйдут из строя 2 детали P(2) = 0.08 = 8.00%**

**в) Вероятность того что в первый месяц выйдут из строя хотя бы одна деталь P(≥1) = 0.46 = 46.00%**

**г) Вероятность того что в первый месяц выйдут из строя от одной до двух деталей P(1,2) = 0.455 = 45.50%**